.github/workflows/ci.yml

```diff
@@ -0,0 +1,208 @@
+      DATABASE_NAME_TEST: Forem_test
+
+    services:
+      postgres:
+        image: postgres:13-alpine
+        env:
+          POSTGRES_PASSWORD: postgres
+        ports:
+          - 5432:5432
+      redis:
+        image: redis
+        ports:
+          - 6379:6379
+
+    strategy:
+      fail-fast: false
+      matrix:
+        ci_node_total: [6]
+        ci_node_index: [0, 1, 2, 3, 4, 5]
+
+    steps:
+      - uses: actions/checkout@v3
+      - name: Cache node modules
+        uses: actions/cache@v3
+        with:
+          path: node_modules
+          key: ${{ runner.os }}-node-modules-${{ hashFiles('**/yarn.lock') }}
+          restore-keys: ${{ runner.os }}-node-modules-
+      - uses: actions/setup-node@v3
+        with:
+          cache: yarn
+      - run: yarn install --frozen-lockfile
+      - name: setup ruby
+        uses: ruby/setup-ruby@v1
+        with:
+          bundler-cache: true
+      - run: bundle exec rails db:test:prepare
+      - name: RSpec
+        run: bin/knapsack_pro_rspec
+      - uses: codecov/codecov-action@v3
+
+  Jest:
+    needs: build-test
+    runs-on: ubuntu-latest
+    timeout-minutes: 20
+    env:
+      RAILS_ENV: test
+      NODE_ENV: test
+
+    steps:
+      - uses: actions/checkout@v3
+      - name: Cache node modules
+        uses: actions/cache@v3
+        with:
+          path: node_modules
+          key: ${{ runner.os }}-node-modules-${{ hashFiles('**/yarn.lock') }}
+          restore-keys: ${{ runner.os }}-node-modules-
+      - uses: actions/setup-node@v3
+        with:
+          cache: yarn
+      - run: yarn install --frozen-lockfile
+      - run: yarn lint:frontend
+      - run: yarn test --colors
+      - uses: codecov/codecov-action@v3
+      - run: yarn build-storybook
+
+  Build-test:
+    runs-on: ubuntu-latest
+    timeout-minutes: 20
+    env:
+      RAILS_ENV: test
+      DATABASE_URL: postgres://postgres:postgres@localhost:5432/Forem_prod_test
+      DATABASE_NAME: Forem_prod_test
+      APP_PROTOCOL: http://
+      APP_DOMAIN: localhost:3000
+      HEROKU_APP_URL: practicaldev.herokuapp.com
+      SECRET_KEY_BASE: dummydummydummy
+      GITHUB_KEY: dummy
+      GITHUB_SECRET: dummy
+      KNAPSACK_PRO_FIXED_QUEUE_SPLIT: true
+
+    services:
+      postgres:
+        image: postgres:13-alpine
+        env:
+          POSTGRES_PASSWORD: postgres
+        ports:
+          - 5432:5432
+      redis:
+        image: redis
+        ports:
+          - 6379:6379
+
+    steps:
+      - uses: actions/checkout@v3
+      - name: Cache node modules
+        uses: actions/cache@v3
+        with:
+          path: node_modules
+          key: ${{ runner.os }}-node-modules-${{ hashFiles('**/yarn.lock') }}
+          restore-keys: ${{ runner.os }}-node-modules-
+      - uses: actions/setup-node@v3
+        with:
+          cache: yarn
+      - run: yarn install --frozen-lockfile
+      - name: setup ruby
+        uses: ruby/setup-ruby@v1
+        with:
+          bundler-cache: true
```

Comment:
**non-blocking question:** Does it make sense to specify a ruby version here? Something along the lines of [this example](https://github.com/ruby/setup-ruby#single-job) on all steps that setup Ruby

At first glance it seems like the action will setup using the latest ruby version. Perhaps this could mean that at some point a new minor/major ruby version will be released and our CI would run on a different release than what we expect. In this case I'm referring to our slightly slower upgrade timelines with Fedora CoreOS. It seems like `.travis.yml` did specify `3.0.2` but I couldn't find any reference to that for GitHub actions here

Reply:
It's omitted on purpose so it can fetch version from our `.ruby-verison`
see https://github.com/forem/forem/actions/runs/3337534541/jobs/5524402013#step:7:17

Reply:
Oh that's so much better! TIL and thanks for clarifying